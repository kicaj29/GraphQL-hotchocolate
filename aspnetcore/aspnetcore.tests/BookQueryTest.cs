using ApprovalTests;
using ApprovalTests.Reporters;
using HotChocolate;
using HotChocolate.Execution;
using Microsoft.Extensions.DependencyInjection;
using NUnit.Framework;
using System;
using System.Collections.Generic;
using System.Text;

namespace aspnetcore.tests
{
    [UseReporter(typeof(DiffReporter))]
    public class BookQueryTest
    {
        [Test]
        public void Test()
        {
            // arrange
            IServiceCollection services = new ServiceCollection();
            var startup = new Startup();
            startup.ConfigureServices(services);
            var provider = services.BuildServiceProvider();
            var executor = provider.GetService<IQueryExecutor>();
            IReadOnlyQueryRequest request = QueryRequestBuilder.New()
                                                .SetQuery(@"
                                                    query {
                                                        books {
                                                            nodes {
                                                                id
                                                                title
                                                                price
                                                            }
                                                        }
                                                    }
                                                ")
                                                .SetServices(provider)
                                                .Create();

            // act
            IExecutionResult result = executor.Execute(request);
            var resultJson = result.ToJson();

            // assert
            Approvals.Verify(resultJson);
        }
    }
}
