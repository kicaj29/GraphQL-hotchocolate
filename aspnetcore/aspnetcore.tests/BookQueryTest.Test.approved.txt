{"data":{"books":{"nodes":[{"id":"1","title":"First Book","price":10},{"id":"2","title":"Second Book","price":11},{"id":"3","title":"Third Book","price":12},{"id":"4","title":"Fourth Book","price":15}]}}}